superset/models/core.py

```diff
@@ -362,6 +362,18 @@ def get_effective_user(self, object_url: URL) -> Optional[str]:
             else None
         )
 
+    @contextmanager
+    def get_sqla_engine_with_context(
```

Comment:
Eventually I think we want to rename this to `get_sqla_engine`, since we want this to be the one and only way to create an engine.